README.md

```diff
@@ -9,8 +9,13 @@ Hint: To make merge sort in-place, think about what happens during the merge --
 where are elements moved to and from? To make it iterative, think about the
 part of the array each recursive call considers.
 
+I used the slides and video from class to look at how merge works and how it sorts. The TA helped me in office hours with runtime analysis. I also used https://www.geeksforgeeks.org/time-and-space-complexity-analysis-of-merge-sort/ to look over the analysis of recursive mergesort. While looking at runtimes for for loops I used https://www.geeksforgeeks.org/how-to-analyse-loops-for-complexity-analysis-of-algorithms/. 
+I certify that I have listed all sources used to complete this exercise, including the use of any Large Language Models. All of the work is my own, except where stated otherwise. I am aware that plagiarism carries severe penalties and that if plagiarism is suspected, charges may be filed against me without prior notice.
+
 ## Runtime Analysis
 
 Analyse the time complexity of your implementation and give a $\Theta$ bound for
 its worst-case runtime. Add your answer, including your reasoning, to this
 markdown file.
+
+In the beginning for loop, we say how big each subarray should be this time around. This will have a divide runtime of log2(n), since in mergesort, the size gets cut from 1 and moves up until it is at the array length moving up by 2s, first "subarray" will be 1, 2, 4, 8...etc. Since there are 3 for loops after this, this makes the runtime $O(n^3)$. This then would come out to $\Theta(n^3log2n)$.
```

Comment:
Are you sure about the complexity? It may be easier to analyze what's different to the normal mergesort.

Reply:
I looked at the merge sort code from the slides and understood where I was messing up my runtime analysis, thank you.